CSVSource/README.md

```diff
@@ -0,0 +1,158 @@
+*   **maxLinesInEvent**: Required. Determine how many lines from the CSV file will be sent in a single message to the server. Depending on the number of the lines of the CSV file, a high value might result in messages too large to process efficiently or a memory exception. 
+*   **delimiter**: the delimiter to be used when parse the CSV file , default is "," , the system will step over null values which might be in the result of the split operation . 
+
+### Schema Configuration
+Schema can be used to control the field names on the uploaded event. If no name is assigned, 'fieldX' will be used where 'X' is the index of the field in the line.  For example field0, field1, etc. 
+
+Those values can be used to override that default behavior for example, the following 
+
+will cause the attribute of the 3rd value in each line to be called "address". 
+So instead of loading event : 
+
+`{ field0:1, field1:true, field2:"there"}` it will upload `{ field0:1, field1:true, address:"there"}`
+this can save conversion processing on the server.
+
+### Execution Options
+
+*   **maxActiveTasks**: Optional. The maximum number of threads running at any given point. This is the number of CSV files being processed simultaneously. Must be a positive integer. Default value is 5.
+*   **maxQueuedTasks**: Optional. The maximum number of queued tasks at any given point for CSV files,overflowing that number might cause lack of data . Must be a positive integer. Default value is 10.
+*   **processExistingFiles**: Optional. If set to `true`, the service will process all files already existing in the folder `fileFolderPath` (filtered using `filePrefix` and the `fileExtension`). Otherwise the service will process only new files.  Default is`false`.
+*   **extensionAfterProcessing**: Optional. Rename the file after it has been processed to avoid reprocessing (_e.g._ for cases where `processExistingFiles` is set to `true`).  The default value is combination of the 'fileExtension' and `done`.  For example `.csv.done` when `fileExtension` set to `.csv`.
+*   **deleteAfterProcessing**: Optional. Delete the processed file only if processed successfully to avoid reprocessing in cases where `processExistingFiles` is set to `true`. Default value is `false`
+
+## Messages from the Source
+
+Messages that are sent to the source as Notifications are JSON objects in the following format:
+```
+{
+   {"field0":columnValue, "field1":columnValue, etc..}
+}
+```
+The data is formatted as a HashMap which represents a row of data. Each map is a series of key-value pairs with the keys 
+being the column name and the values being the column value. If multiple rows of data are returned by the notification, the number iof max events is determined by the `maxLinesInEvent` 
+
+## Event structure
+Each event consists on the follwoing structure :
+
+* **file** : the source file from where the data was extracted
+* **segment** : number of CSV lines on the current segment 
```

Comment:
by _segment_, I assume you mean the number of lines processed to create the segment contained in the current event.  Specifically, in the `lines` property?  It would be good to pick a single set of terminology to use.  Maybe `filename`, `lineCount`, and `segment`.  By the time they get here, they aren't "lines" anymore -- the data is an array of JSON objects whose content was extracted from some lines in the file.

Reply:
i have update the meaning of segment . no points to send number of line in the message as it exist in the array .